<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OCC::ThreatHistoryDialog</class>
 <widget class="QDialog" name="OCC::ThreatHistoryDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Threat History</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="toolbarLayout">
     <item>
      <widget class="QLabel" name="periodLabel">
       <property name="text">
        <string>Period:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="periodComboBox">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Filter threats by time period</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="toolbarSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="exportButton">
       <property name="text">
        <string>Export CSV</string>
       </property>
       <property name="toolTip">
        <string>Export threat history to CSV file</string>
       </property>
       <property name="icon">
        <iconset theme="document-save"/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="clearButton">
       <property name="text">
        <string>Clear History</string>
       </property>
       <property name="toolTip">
        <string>Clear all threat history (cannot be undone)</string>
       </property>
       <property name="styleSheet">
        <string>QPushButton { color: #DC3545; }</string>
       </property>
       <property name="icon">
        <iconset theme="edit-delete"/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="threatTable">
     <property name="styleSheet">
      <string>QTableWidget {
    font-family: system-ui, -apple-system, sans-serif;
}
QTableWidget::item {
    padding: 4px;
}</string>
     </property>
     <property name="alternatingRowColors">
      <bool>true</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="showGrid">
      <bool>true</bool>
     </property>
     <property name="gridStyle">
      <enum>Qt::SolidLine</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
     <attribute name="horizontalHeaderVisible">
      <bool>true</bool>
     </attribute>
     <attribute name="horizontalHeaderCascadingSectionResizes">
      <bool>false</bool>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="hintLabel">
     <property name="text">
      <string>Double-click a threat to view details</string>
     </property>
     <property name="styleSheet">
      <string>color: gray; font-style: italic;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="footerLayout">
     <item>
      <widget class="QLabel" name="statsLabel">
       <property name="text">
        <string>0 threats</string>
       </property>
       <property name="styleSheet">
        <string>font-weight: bold;</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="footerSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="closeButton">
       <property name="text">
        <string>Close</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
